pkg/provider/gke/gke.go

```diff
@@ -77,6 +77,10 @@ type GKE struct {
 
 // NewGKEClient sets the GKE client used when performing GKE requests.
 func (c *GKE) NewGKEClient(*kingpin.ParseContext) error {
+
+	// Set PROVIDER_SERVICE Deployment variable to  LoadBalancer
+	c.DeploymentVars["PROVIDER_SERVICE"] = "LoadBalancer"
```

Comment:
```suggestion
	c.DeploymentVars["NGINX_SERVICE_TYPE"] = "LoadBalancer"
```

In the provider specific docs, we could just add a section about what are the deployment variables that are automatically passed for this specific provider. 

It'll be good if we could have these deployment variables overridable when uses passes the same using the `-v` flag. Maybe also have a provider specific flag to list out all the set deployment variables. (This way we won't have to update readme manually and we could just use embedmd https://github.com/prometheus/test-infra/pull/353)

What I think will be good is to have a default `DeploymentVars` and have it set to the default value, like for eg. in this example only nginx controller k8s service type needs to be changed when running in KIND.

But for EKS and GKE, the type `LoadBalancer` works, so we could have some defaults and only modify them when creating the provider that needs the change(in our case KIND)